packages/core/src/oidc/init.ts

```diff
@@ -114,6 +114,19 @@ export default async function initOidc(app: Koa): Promise<Provider> {
         return refreshTokenTtl ?? defaultRefreshTokenTtl;
       },
     },
+    extraTokenClaims: async (ctx, token) => {
+      // AccessToken type is not exported by default, need to asset token is AccessToken
+      if (token.kind === 'AccessToken') {
+        const { accountId } = token;
+        const { roleNames } = await findUserById(accountId);
+
+        // Add User Roles to the AccessToken claims. Should be removed once we have RBAC implemented.
+        // User Roles should be hidden and  determined by the AccessToken scope only.
+        return {
+          roleNames,
```

Comment:
This would end up in the `idTokenClaims`, right? So it's the thing we talk about yesterday again, we need to synchronize and update  the `IdTokenkClaims` typescript type in SDKs

Reply:
nope, this is for access_token use

Reply:
OK, but from the screenshot, this looks like idTokenClaims to me

<img width="327" alt="image" src="https://user-images.githubusercontent.com/12833674/169734238-559863be-29b0-472d-a2a5-aee938540595.png">

Reply:
yes, they are quite similar as they all use JWT format, and share some common JWT params. 

As idToken is returned and consumed only on authentication request flow.  
But access token, on the other hand,  is used to authorize all the following incoming requests. So we need to add this roleName claims to the access token and validate in each koaAuth API route. 

Also, this `extraTokenClaims` OIDC config is used to put extra claims to the acess_token only.  We  use`claims` and `findAccount` to customize idToken claims.

Reply:
OK, thanks for the detailed explanation 👍